internal/unionstore/art/arena.go

```diff
@@ -0,0 +1,358 @@
+package art
+
+import (
+	"encoding/binary"
+	"math"
+	"sync/atomic"
+	"unsafe"
+
+	"github.com/tikv/client-go/v2/internal/logutil"
+	"go.uber.org/zap"
+)
+
+const (
+	alignMask = 1<<32 - 8 // 29 bit 1 and 3 bit 0.
```

Comment:
I copied it from [memdb's arena](https://github.com/tikv/client-go/blob/cd64e24de8d1a4edb9852930fdeddc4abbe07692/internal/unionstore/memdb_arena.go#L49).
My understanding is because [the address](https://github.com/tikv/client-go/blob/cd64e24de8d1a4edb9852930fdeddc4abbe07692/internal/unionstore/memdb_arena.go#L63) is 32bit integer, 32 bit mask is enough.

Reply:
My point is will it overflow in a 32-bit machine?

Reply:
However, modifying the type would necessitate changes in numerous locations where int is currently implemented. In my opinion, we should avoid using `int` in these places... I'm comfortable with simply declaring that we don't support 32-bit systems.

Reply:
> My point is will it overflow in a 32-bit machine?

The `maxBlockSize` is set to 128MB(128 << 20), which is far below maxuint32. The offset of address is less than `maxBlockSize` so it won't overflow.

----

We can declare it don't support 32-bit systems, but I still want to use uint32 address 1 it reduces the memory usage for address. Is `const alignMask uint32 = 0xFFFFFFF8` what you mean by avoiding using `int`.